src/types/simple/serialize.rs

```diff
@@ -32,51 +34,61 @@ pub(crate) enum SerSimpleType {
     },
     List {
         inner: Box<SimpleType>,
-        l: bool,
+        c: TypeClass,
     },
     Map {
         k: Box<SerSimpleType>,
         v: Box<SerSimpleType>,
-        l: bool,
+        c: TypeClass,
     },
     Tuple {
         row: Box<TypeRow<SerSimpleType>>,
-        l: bool,
+        c: TypeClass,
     },
     Sum {
         row: Box<TypeRow<SerSimpleType>>,
-        l: bool,
+        c: TypeClass,
     },
     Array {
         inner: Box<SerSimpleType>,
         len: usize,
-        l: bool,
+        c: TypeClass,
     },
     Opaque {
         custom: CustomType,
-        l: bool,
+        c: TypeClass,
     },
     Alias {
         name: SmolStr,
-        l: bool,
+        c: TypeClass,
     },
     Var {
         name: SmolStr,
     },
 }
 
-impl PrimType for SerSimpleType {}
+impl PrimType for SerSimpleType {
+    fn class(&self) -> TypeClass {
+        unimplemented!()
```

Comment:
I don't think this should ever be called (we actually only use CLASS in this file, and instances of SerSimpleType should not be seen outside this file)

Reply:
But this was one reason for keeping CLASS - we can (statically) avoid the need for `SerSimpleType::CLASS` and then duck out of this one dynamically